### PR TITLE
feat(build-common): Add new shared API-Extractor configuration that enables type-trimming

### PR DESCRIPTION
**Pull request by Josmithr**
Replaces the shared [API-Extractor](https://api-extractor.com/) configurations in `build-common` with a single, strict configuration that also enables type-trimming for `alpha`, `beta`, and `public` typings.

The new config captures our stricter ruleset, and also adds the `explicit-release-tag` requirement. Packages will either need to fix violations when adopting, or explicitly (and _temporarily_) opt out of those rules in their local configs when integrating this new config.

[AB#5693](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5693)

### CODE REVIEW

**Comment by Josmithr** [comment]
PR updating api-markdown-documenter to require explicit release tags: #17258

File: configs/api-extractor-base.json
Comment (tylerbutler) [inline configs/api-extractor-base.json:1]: nit: we don't have a convention of putting things in a configs folder, and adding a general-purpose one makes me wonder what goes in there vs the root? Maybe name this `api-extractor`configs`?

Comment (Josmithr) [reply configs/api-extractor-base.json:1]: I was going to suggest starting to organize shared config bases in such a directory as a means to declutter the root a bit. But this file could also just live at the root.

Comment (Josmithr) [reply configs/api-extractor-base.json:1]: I went ahead and put the config back in `build-common` for now. We can tackle extracting configs from that package separately.

**Review by tylerbutler — APPROVED** [review]

File: common/build/build-common/api-extractor-base.json [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+		"enabled": true,
+		"apiJsonFilePath": "<projectFolder>/_api-extractor-temp/doc-models/<unscopedPackageName>.api.json"
+	},
+
+	/**
+	 * Configures how the .d.ts rollup file will be generated.
+	 */
+	"dtsRollup": {
+		"enabled": true,
+		"alphaTrimmedFilePath": "<projectFolder>/dist/<unscopedPackageName>-alpha.d.ts",
```
Comment (Josmithr) [inline common/build/build-common/api-extractor-base.json:39]: @sonalideshpandemsft it just occurred to me that I should check what is the most convenient naming schema from the tooling perspective, as you're working on that side of things. We're not locked into this `<package-name>-<scope>` scheme.

Would it be better to omit the package name from these?

File: common/build/build-common/api-extractor-base.json [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	"docModel": {
+		"enabled": true,
+		"apiJsonFilePath": "<projectFolder>/_api-extractor-temp/doc-models/<unscopedPackageName>.api.json"
+	},
+
+	/**
+	 * Configures how the .d.ts rollup file will be generated.
+	 */
+	"dtsRollup": {
+		"enabled": true,
```
Comment (sonalideshpandemsft) [inline common/build/build-common/api-extractor-base.json:38]: I think we should set the default `dtsRollup - enabled` property to false. This updates alpha/beta types in all the packages. example run: https://dev.azure.com/fluidframework/internal/_build/results?buildId=200523&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=3e8a0811-333e-5612-55d7-750fabab9520

any reason why this is set to true as default?

Comment (Josmithr) [reply common/build/build-common/api-extractor-base.json:38]: There isn't a reason to disable this behavior IMO. Packages can opt out if they want to. But I expect most of our packages will use these, and in the meantime it's harmless to generate them.

**Pull request merged**